### PR TITLE
BIGTOP-4059: Add Documentation for Parallel Maven Compilation in Bigtop

### PR DESCRIPTION
**Pull request by JiaLiangC**
### Description of PR
Add Documentation for Parallel Maven Compilation in Bigtop

### For code changes:

- [ ] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'BIGTOP-3638. Your PR title ...')?
- [ ] Make sure that newly added files do not have any licensing issues. When in doubt refer to https://www.apache.org/licenses/

### CODE REVIEW

**Comment by JiaLiangC** [comment]
@iwasakims Could you help review this PR?

**Comment by JiaLiangC** [comment]
@guyuqi Could you help review this PR?

**Comment by guyuqi** [comment]
+1

File: bigtop.bom [hunk 1]
```diff
@@ -86,7 +86,10 @@
        packaging = 'rpm' // *optional* If this component can be built only as either
                          // DEB or RPM, specify that packaging format explicitly.
                          // If both formats are supported, omit this option.
-       maven_parallel_build = true
+       maven_parallel_build = true // *optional* maven_parallel_build indicates whether the component can be built in parallel.
+                                   // Components with this flag can have the Maven parallel build parameter -PbuildThreads=2C or -PbuildThreads=2
```
Comment (iwasakims) [inline bigtop.bom:90]: Currently `-PbuildThread=2` is not allowed. How about fix [the code](https://github.com/apache/bigtop/blob/fc93b76a8d3f51531af730f9f6397e91fd6b91cb/packages.gradle#L456-L457) here? @JiaLiangC

Comment (iwasakims) [reply bigtop.bom:90]: Oops. My comment conflicts with @guyuqi. We can fix this with follow-up JIRA.

Comment (JiaLiangC) [reply bigtop.bom:90]: > Currently `-PbuildThread=2` is not allowed. How about fix [the code](https://github.com/apache/bigtop/blob/fc93b76a8d3f51531af730f9f6397e91fd6b91cb/packages.gradle#L456-L457) here? @JiaLiangC

@iwasakims  Thank you for the reminder. I will fix the code.

**Pull request merged**